lib/peer-set.js

```diff
@@ -0,0 +1,25 @@
+// when we have to dupe a set of peer dependencies deeper into the tree in
+// order to make room for a dep that would otherwise conflict, we use
+// this to get the set of all deps that have to be checked to ensure
+// nothing is locking them into the current location.
+//
+// this is different in its semantics from an "optional set" (ie, the nodes
+// that should be removed if an optional dep fails), because in this case,
+// we specifically intend to include deps in the peer set that have
+// dependants outside the set.
+const peerSet = node => {
+  const set = new Set([node])
+  for (const node of set) {
```

Comment:
ah curious, can't it just get rid of this initial `for` loop and just use the provided `node` param directly? or maybe it was meant to accept arrays in the param and it should use `const set = new Set([...node])` instead?

Reply:
The set is initialized to `new Set([node])` but then we add to the set in the loop, so it gets longer.

Ie, we start with 1 node, then if it has peer deps, we add those to the set.  Using a set rather than building up an array so that we get deduplication for free.